tachyon/base/files/memory_mapped_file_posix.cc

```diff
@@ -0,0 +1,147 @@
+// Copyright 2013 The Chromium Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "tachyon/base/files/memory_mapped_file.h"
+
+#include <fcntl.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include "tachyon/base/files/file_util.h"
+#include "tachyon/base/logging.h"
+#include "tachyon/base/numerics/safe_conversions.h"
+// #include "tachyon/base/threading/scoped_blocking_call.h"
+#include "tachyon/build/build_config.h"
+
+namespace tachyon::base {
+
+MemoryMappedFile::MemoryMappedFile() = default;
+
+#if !BUILDFLAG(IS_NACL)
+bool MemoryMappedFile::MapFileRegionToMemory(
+    const MemoryMappedFile::Region& region,
+    Access access) {
+  // TODO(chokobole):
+  // ScopedBlockingCall scoped_blocking_call(FROM_HERE, BlockingType::MAY_BLOCK);
+
+  off_t map_start = 0;
+  size_t map_size = 0u;
+  int32_t data_offset = 0;
+  size_t byte_size = 0u;
+
+  if (region == MemoryMappedFile::Region::kWholeFile) {
+    int64_t file_len = file_.GetLength();
+    if (file_len < 0) {
+      DPLOG(ERROR) << "fstat " << file_.GetPlatformFile();
+      return false;
+    }
+    if (!IsValueInRangeForNumericType<size_t>(file_len))
+      return false;
+    map_size = base::checked_cast<size_t>(file_len);
+    byte_size = map_size;
+  } else {
+    // The region can be arbitrarily aligned. mmap, instead, requires both the
+    // start and size to be page-aligned. Hence, we map here the page-aligned
+    // outer region [|aligned_start|, |aligned_start| + |size|] which contains
+    // |region| and then add up the |data_offset| displacement.
+    int64_t aligned_start = 0;
+    size_t aligned_size = 0u;
+    CalculateVMAlignedBoundaries(region.offset,
+                                 region.size,
+                                 &aligned_start,
+                                 &aligned_size,
+                                 &data_offset);
+
+    // Ensure that the casts in the mmap call below are sane.
+    if (aligned_start < 0 ||
+        !IsValueInRangeForNumericType<off_t>(aligned_start)) {
+      DLOG(ERROR) << "Region bounds are not valid for mmap";
+      return false;
+    }
+
+    map_start = base::checked_cast<off_t>(aligned_start);
+    map_size = aligned_size;
+    byte_size = region.size;
+  }
+
+  if (map_size == 0u) {
+    // mmap() requires `map_size > 0`, and this ensures an empty span indicates
```

Comment:
Is this taken from Chromium (aka we should leave it alone? Else, please wrap names with "|")

Reply:
Yes it was taken from chromium! is it okay to keep it as it is for maintenance?